crates/net/network/src/network.rs

```diff
@@ -211,4 +211,6 @@ pub(crate) enum NetworkHandleMessage {
     FetchClient(oneshot::Sender<FetchClient>),
     /// Apply a status update.
     StatusUpdate { height: u64, hash: H256, total_difficulty: U256 },
+	/// Get PeerInfo
+	GetPeerInfo,
```

Comment:
the way this should work is similar to `FetchClient` command. It should take a `oneshot::Sender` through which the `NetworManager` will send the response. The `Handle` can await an the receiver half of the channel and return the response asynchronously.